### ISSUE TITLE
Inquiry About OLMo 2 1B DPO and 32B RLVR Configurations

### ISSUE THREAD

**Issue by LuckyDaydreamer** [allenai/open-instruct#1306]
Hi Team,

Great work! Could you please share the configuration files for OLMo 2 1B DPO and OLMo 2 32B RLVR? I wasn’t able to find them under configs/train_configs/olmo2, scripts/train/olmo2, or in the issues posted by other users.

Thank you!

Best,
Tao

**Comment by BrownianNotion** [comment]
Hi Tao,
I've also been looking at reproducing the OLMo-2 1B results. The configs are under `docs/olmo2.md` [here](https://github.com/allenai/open-instruct/blob/main/docs/olmo2.md), but I believe there are a few errors (eg. tokenizer links to 13B model). This is what I've managed to reconcile for DPO. 

Note that if you have multiple GPUs, each with 80GB of memory, you probably won't need deep speed for the 1B model, I suggest creating your own stage 0 config - this sped up the SFT stage by above 30% for me.

```
accelerate launch \
    --mixed_precision bf16 \
    --num_processes 8 \
    --use_deepspeed \
    --deepspeed_config_file configs/ds_configs/stage2_accelerate.conf \
    --deepspeed_multinode_launcher standard \
    open_instruct/dpo_tune_cache.py \
    --exp_name 0424_1B_dpo_onpol_lr_2.5e-6_seed_111 \
    --learning_rate 2.5e-6 \
    --seed 111 \
    --model_name_or_path allenai/OLMo-2-0425-1B-SFT \
    --model_revision main \
    --use_flash_attn \
    --tokenizer_name_or_path allenai/OLMo-2-0425-1B-SFT \
    --tokenizer_revision main \
    --max_seq_length 2048 \
    --chat_template_name tulu \
    --per_device_train_batch_size 2 \
    --gradient_accumulation_steps 8 \
    --lr_scheduler_type linear \
    --warmup_ratio 0.1 \
    --weight_decay 0.0 \
    --num_train_epochs 1 \
    --output_dir /output \
    --with_tracking \
    --report_to wandb \
    --logging_steps 1 \
    --gradient_checkpointing \
    --dataset_mixer_list allenai/olmo-2-0425-1b-preference-mix 1.0 \
    --use_slow_tokenizer False \
    --add_bos \
    --use_lora False \
    --dpo_loss_type dpo_norm \
    --dpo_beta 5 
```

**Comment by LuckyDaydreamer** [comment]
Thank you @BrownianNotion! This has been super helpful.

Best,
Tao

**Comment by BrownianNotion** [comment]
FYI: Deepspeed stage 2 has a bug that causes nan loss for DPO: https://github.com/deepspeedai/DeepSpeed/issues/7188 . This was fixed in `0.17.5` but the docker image currently uses `0.15.4`.

All the more reason to switch to stage 0 where appropriate (otherwise just turn off either `overlap_comm` or `contiguous_gradients`).

**Comment by hamishivi** [comment]
Hi, the configs pointed out are correct! We just pointed the tokenizer to the 13b instance for... random reasons I guess haha.
For the 32b, you can look at the huggingface page [here](https://huggingface.co/allenai/OLMo-2-0325-32B-Instruct#reproduction-command)!

**Comment by BrownianNotion** [comment]
> FYI: Deepspeed stage 2 has a bug that causes nan loss for DPO: [deepspeedai/DeepSpeed#7188](https://github.com/deepspeedai/DeepSpeed/issues/7188) . This was fixed in `0.17.5` but the docker image currently uses `0.15.4`.
> 
> All the more reason to switch to stage 0 where appropriate (otherwise just turn off either `overlap_comm` or `contiguous_gradients`).

Hopefully won't be a problem anymore after #1311 , thanks team!
